packages/code-studio/src/settings/SettingsMenu.tsx

```diff
@@ -177,7 +193,20 @@ export class SettingsMenu extends Component<
                 </a>
               </div>
               <div className="app-settings-footer-item">
-                <div className="text-muted">Version {version}</div>
+                <div className="text-muted">
+                  Engine Version {deephavenVersion}{' '}
+                  <div className="detailed-server-config">
+                    <FontAwesomeIcon icon={vsInfo} />
+                    <Tooltip interactive>
+                      <div className="detailed-server-config-info">
+                        {getRow('UI Version', uiVersion)}
+                        {getRow('Deephaven Version', deephavenVersion)}
+                        {getRow('Java Version', javaVersion)}
+                        {getRow('Barrage Version', barrageVersion)}
+                      </div>
+                    </Tooltip>
+                  </div>
+                </div>
```

Comment:
```suggestion
                <div className="font-weight-bold">Version</div>
                <span className="d-inline-block text-muted">
                  {deephavenVersion} <FontAwesomeIcon icon={vsInfo} />
                  <Tooltip interactive>
                    <div className="detailed-server-config">
                      {getRow('Engine Version', deephavenVersion)}
                      {getRow('Web UI Version', uiVersion)}
                      {getRow('Java Version', javaVersion)}
                      {getRow('Barrage Version', barrageVersion)}
                    </div>
                  </Tooltip>
                </span>
```

Re-styled a bit, and adjusted the order/labels.